indico/modules/events/registration/util.py

```diff
@@ -457,3 +440,37 @@ def generate_ticket_qr_code(registration):
     qr.add_data(json_qr_data)
     qr.make(fit=True)
     return qr.make_image()._img
+
+
+def get_open_or_registered_regforms(event, user):
+    """Get registration forms of an event which are open or user has already registered to.
+
+    :param event: the `Event` to get registrations for
+    :param user: A `User`
+    """
+    if not event.has_feature('registration'):
+        return []
+
+    return (RegistrationForm.query.with_parent(event)
+            .filter((RegistrationForm.is_open) |
+                    (RegistrationForm.registrations.any((Registration.user == user) & (~Registration.is_deleted))))
+            .order_by(db.func.lower(RegistrationForm.title))
+            .all())
+
+
+def get_registration_counts_dict(event, publish=False):
+    """Get dictionary with number of registrations for each registration form of the event
```

Comment:
`s/dictionary/dict/` and that line is a big long; usually  docstring lines shouldn't  be longer than 72 chars.